publication-rest/src/main/java/no/unit/nva/publication/delete/DeletePublicationHandler.java

```diff
@@ -42,14 +45,32 @@ public DeletePublicationHandler(ResourceService resourceService,
         super(Void.class, environment);
         this.resourceService = resourceService;
         this.identityServiceClient = identityServiceClient;
+        this.publicationPermissionStrategy = new PublicationPermissionStrategy();
     }
 
     @Override
     protected Void processInput(Void input, RequestInfo requestInfo, Context context) throws ApiGatewayException {
-        SortableIdentifier identifier = RequestUtil.getIdentifier(requestInfo);
         var userInstance = createUserInstanceFromRequest(requestInfo);
-        
-        resourceService.markPublicationForDeletion(userInstance, identifier);
+        var publicationIdentifier = RequestUtil.getIdentifier(requestInfo);
+
+        var publication = resourceService.getPublicationByIdentifier(publicationIdentifier);
+        var publicationStatus = publication.getStatus();
+
+        switch (publicationStatus) {
+            case PUBLISHED:
+                if (!publicationPermissionStrategy.hasPermissionToUnpublish(requestInfo, publication)) {
+                    throw new UnauthorizedException();
+                }
```

Comment:
If using constructor, in my opinion it would be more readable to name this something like this:

UserPermissions.fromRequestInfo(requestInfo).hasPermissionToUnpublishPublication(publication)

But it is up to you :)